### PR TITLE
feat(tests): suites lead with Run, and the suite's tests are a grid

### PR DESCRIPTION
**Pull request by thebhulawat**
Ticket 02 of the agents-tests-rework effort (planning .scratch/agents-tests-rework/issues/02).

## What changed
- **Suites screen**: Run a suite primary (far right), Create suite secondary; columns exactly Name · Created; row menu exactly Rename · Run suite · Delete suite; Copy suite id removed with no home anywhere; empty state unchanged and leads with creation.
- **The suite's tests screen is a spreadsheet grid** (new tests-grid.tsx): Name · Scenario · Expected behaviors · Personas. Cell-by-cell saves (blur/Enter commit ONE field with the version guards; Escape cancels; emptying a mandatory field refused in place with one sentence). Entry-row creation with a pinned Save test/Cancel bar — Save disabled with a true missing-fields sentence until name, scenario, ≥1 behavior, ≥1 persona exist; one click = one createTest; centred discard dialog for typed content. Add-affordances only on the woken cell; the persona picker lives only in the cell. Empty suite = header + teaching ghost row.
- **Retired**: the test full page and the write-a-test sheet. /tests/:testId resolves the suite and lands on its grid; /tests/new?suite= opens the grid's entry row; /tests/new bare lands on the suites screen.
- **Contract change**: createTest requires ≥1 persona; the silent default-persona substitution is replaced by a refusal ("a test needs at least one persona, because a test says who calls") on create and on edits that would empty personas. Regenerated SDK/types/OpenAPI in the commit; nine api/db test files moved with the rule.

## Known v0 gaps, recorded for the founder (deliberate, not bugs)
- The description field and mock-overrides read view are unreachable from the UI (their only home was the retired test page; fields survive in API/DB).
- A refused cell save prints the platform's 409 sentence in place; the old copy-draft/read-again pair went with the page.
- A fix commit lands shortly on this branch restoring **Delete test** as the grid rows' ⋮ action (the silence rule keeps today's verbs).

## Tests (scoped)
- apps/web/test/test-suites-pages.test.tsx — 16 passed · packages/db tests (tests, tests-edit-versions, personas-named-by-tests, tests-behaviors) — 67 passed · apps/api (test-suites-cutover, runs-suite-contract, mock-tools-world, project-context, agents-lifecycle) — 63 passed · pnpm typecheck green. Full lanes run here in CI.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
<a href="https://app.blacksmith.sh/egma-ai/codesmith/egma/pr/217"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with [code]smith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1790240290&installation_model_id=431960&pr_number=217&repository=egma-ai%2Fegma&return_to=https%3A%2F%2Fgithub.com%2Fegma-ai%2Fegma%2Fpull%2F217&signature=7e8c621e36a7b02fe251229d7302eccaef2a41a274e36a137cb3bde6d4cf264f"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with [code]smith</picture></a>
<sup>Need help on this PR? Tag <code>@codesmith-bot</code> with what you need. Autofix is disabled.</sup>

<h3>Greptile Summary</h3>

The PR reorganizes test suites around direct suite execution and replaces individual test editing pages with an inline spreadsheet-style grid.
- Adds guarded, field-level test editing and an inline test-creation row.
- Redirects retired test and test-creation routes into the corresponding suite grid.
- Requires every test to name at least one persona across the API, database, web, and CLI flows.
- Updates generated platform contracts and the affected integration tests.

<h3>Confidence Score: 4/5</h3>

The PR is not yet safe to merge because committing cell B while cell A is still saving silently discards B's save action.

The follow-up correctly prevents a late completion from clearing the newly active cell, but the shared saving guard still returns before issuing or queuing a second cell commit, so overlapping edits can remain visibly drafted without being persisted.

**Files Needing Attention:** apps/web/app/projects/[projectId]/tests/tests-grid.tsx

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| apps/web/app/projects/[projectId]/tests/tests-grid.tsx | Introduces the inline tests grid and hardens late-save ownership checks, but the shared in-flight guard still drops a second cell commit. |
| apps/web/app/projects/[projectId]/tests/[testId]/page.tsx | Replaces the retired detail page with a project-scoped lookup and redirect to the test's suite. |
| apps/web/app/projects/[projectId]/tests/new/page.tsx | Redirects legacy test-creation links to the suite grid with its entry row open. |
| packages/db/src/access/tests.ts | Enforces the new invariant that tests retain at least one persona during creation and edits. |
| apps/cli/src/wizard/generate-step.ts | Reads project personas before generation so generated test files satisfy the new persona requirement. |
| packages/platform-api/src/contract/operations/tests.ts | Updates the public test contract to require at least one persona. |

<details open><summary><h3>Flowchart</h3></summary>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart LR
  Suites[Suites screen] --> Grid[Suite tests grid]
  OldTest[Retired test URL] --> Grid
  OldNew[Retired new-test URL] --> Entry[Grid entry row]
  Entry --> Create[Create test with persona]
  Grid --> Patch[Cell-level guarded update]
```
</details>

<sub>Reviews (4): Last reviewed commit: ["fix(wizard): generate files that name a ..."](https://github.com/egma-ai/egma/commit/4e3e7f2a67f852541af3a2d83b3617219d78d245) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=56583334)</sub>

### CODE REVIEW

File: apps/web/app/projects/[projectId]/tests/tests-grid.tsx [hunk 1]
```diff
@@ -0,0 +1,955 @@
...
+    if (answer.status === "signed-out") {
+      window.location.replace("/sign-in");
+      return;
+    }
+    if (answer.status !== "ready") {
+      setCellRefused(answer.refusal.message);
+      return;
+    }
+    onSaved(answer.value);
+    rest();
```
[the remark this answers is not in the document]
Comment (thebhulawat) [reply apps/web/app/projects/[projectId]/tests/tests-grid.tsx]: Fixed in e9728d61: commit() captures which cell the save belongs to and the completion only rests that cell — a save landing after the person moved on no longer touches the newer cell's wake state or draft. A follow-up commit on this branch hardens the compare to live state (functional updates) per the newer thread below, with a race test covering commit-A → edit-B → A-resolves.

File: apps/web/app/projects/[projectId]/tests/tests-grid.tsx [hunk 2]
```diff
@@ -0,0 +1,1141 @@
...
+   * throw away what was typed with no refusal and no record — the one thing
+   * this grid promises never to do.
+   */
+  function rest(mine?: { readonly testId: string; readonly field: Field }): void {
+    if (
+      mine !== undefined &&
+      (active?.testId !== mine.testId || active.field !== mine.field)
+    ) {
+      return;
+    }
```
[the remark this answers is not in the document]
Comment (thebhulawat) [reply apps/web/app/projects/[projectId]/tests/tests-grid.tsx]: Confirmed and fixed in 4e3e7f2a — the challenge was right: rest(mine) compared against the render-closure active, so a late answer for A could still clear B. The comparison now reads a ref written in the same breath as the state (wokenNow), on both the success and refusal paths; the double-submit guard moved to a ref the same way. The new race test holds A's PATCH open, edits B, releases A, and asserts B keeps both its wake and its typed draft — and it was proven to catch the bug: with the one line reverted to the closure read, the test fails.

File: apps/web/app/projects/[projectId]/tests/tests-grid.tsx [hunk 3]
```diff
@@ -0,0 +1,955 @@
...
+    });
+    return () => {
+      live = false;
+    };
+  }, [projectId]);
+
+  const wanted = search.trim().toLocaleLowerCase();
+  const listed = (people ?? []).filter(
+    (one) => wanted === "" || one.name.toLocaleLowerCase().includes(wanted),
+  );
```
[the remark this answers is not in the document]
Comment (thebhulawat) [reply apps/web/app/projects/[projectId]/tests/tests-grid.tsx:287]: Fixed in 4e3e7f2a. The picker now follows nextPageToken to exhaustion (bounded, rendering each page as it lands), so later-page personas are listed, searchable, and pickable — proven with a two-page stub whose page-two persona lands in the createTest body. The truncation note was also corrected to be honest (client-side search reaches only what was read); server-side search is the recorded follow-up for very large rosters.
